### ISSUE TITLE
add tags or roles for different server capabilities

### ISSUE THREAD

**Issue by MartinX3** [MartinX3-AdministrativeDevelopment/ansible-archlinux-encrypted-root#30]
Different server cababilities allows different feature-sets
If we assume that the basic server has less than 6GB ram, on top of the basic features a 6GB ram server could handle:
- `~/.cache` in `tmpfs`

If we assume that the basic server has AES-NI, but not a different server:
- use `serpent-xts` with `512b`

**Comment by Loader009** [comment]
>     * `~/.cache` in `tmpfs`
variable flag -> RAM_for_caching ?
 
>     * use `serpent-xts` with `512b`
This really depends on the intention.
Some (paranoid) might consider that aes shall never be used - thus I let them decide all parameters with variables. 

>     * Add uefi (partition) support with `systemd-boot` and `xbootldr` partition and it's hook
dupe -> https://github.com/MartinX3-AdministrativeDevelopment/ansible-archlinux-encrypted-root/issues/2

**Comment by MartinX3** [comment]
Yes, documented variables with listed parameters I could insert would be nice.

**Comment by MartinX3** [comment]
About the cache change

```bash
# /etc/environment
#
# This file is parsed by pam_env module
#
# Syntax: simple "KEY=VAL" pairs on separate lines
#

# https://wiki.archlinux.org/index.php/XDG_Base_Directory
# .cache should be in tmpfs in the RAM
XDG_CACHE_HOME="/tmp/${USER}/.cache"
```
and
```bash
# /etc/profile.d/xdg-base-directory.sh
# https://wiki.archlinux.org/index.php/XDG_Base_Directory
# .cache should be in tmpfs in the RAM
if [ $USER ]; then
  export XDG_CACHE_HOME="/tmp/${USER}/.cache"
fi

```

**Comment by Loader009** [comment]
Isn't this pretty complicated and it could also make issues with programs which have .cache hardcoded? (yes - I know, the correct way is to use the variable)

My way would be to mount .cache as tmpfs, without touching any variable.

**Comment by MartinX3** [comment]
Hardcoded apps still use `~/.cache/`which isn't forbidden, so no error would happen.

**Comment by Loader009** [comment]
then maybe a symlink from `~/.cache` to `/tmp/${USER}/.cache`?

**Comment by MartinX3** [comment]
Maybe this solution was so easy, that I didn't try it or this didn't work because after a restart the path `/tmp/${USER}/.cache` doesn't exist.
